builders/builder.go

```diff
@@ -0,0 +1,47 @@
+package builders
+
+import (
+	"github.com/fossas/fossa-cli/config"
+	"github.com/fossas/fossa-cli/module"
+)
+
+// A Builder is an implementation of functionality for different build systems.
+type Builder interface {
+	// Initialize gathers build environment information and does build setup.
+	Initialize() error
+	// Build runs a best-effort attempt at building the module.
+	Build(m module.Module, force bool) error
+	// Analyze returns the dependencies of a module.
+	Analyze(m module.Module, allowUnresolved bool) ([]module.Dependency, error)
+
+	// IsBuilt checks whether a module has been built.
+	IsBuilt(m module.Module, allowUnresolved bool) (bool, error)
+
+	// IsModule takes a configKey and returns whether it is valid to elect this Builder.
```

Comment:
What is a `configKey`? What is electing, and why is that different from choosing?

If you're going to use weird vocabulary, you should define it because others will assume there is a specific definition.